stac_fastapi/types/stac_fastapi/types/core.py

```diff
@@ -341,7 +342,20 @@ class BaseCoreClient(LandingPageMixin, abc.ABC):
         factory=lambda: BASE_CONFORMANCE_CLASSES
     )
     extensions: List[ApiExtension] = attr.ib(default=attr.Factory(list))
-    post_request_model = attr.ib(default=BaseSearchPostRequest)
+    post_request_model = attr.ib(default=None)
+
+    @post_request_model.validator
+    def _deprecate_post_model(self, attribute, value):
+        """Check and raise warning if `post_request_model` is set."""
+        if value is not None:
+            warnings.warn(
+                "`post_request_model` attribute is deprecated and will be removed in 3.1",
+                DeprecationWarning,
+            )
+
+    def __attrs_post_init__(self):
+        """Set default value for post_request_model."""
+        self.post_request_model = self.post_request_model or BaseSearchPostRequest
```

Comment:
this might be a breaking change after all 😬 

I need to be able to set this as default but also make sure we can raise a warning if users are setting it 

🤔